kornia/feature/harris.py

```diff
@@ -74,42 +85,70 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
                              .format(input.shape))
         # compute the first order gradients with sobel operator
         # TODO: implement support for kernel different than three
-        gradients: torch.Tensor = spatial_gradient(input)
+        grad_mode = 'Sobel' if self.do_blur else 'NoBlur'
+        gradients: torch.Tensor = spatial_gradient(input, grad_mode)
         dx: torch.Tensor = gradients[:, :, 0]
         dy: torch.Tensor = gradients[:, :, 1]
 
         # compute the structure tensor M elements
         def g(x):
             return gaussian_blur2d(x, (3, 3), (1., 1.))
-
         dx2: torch.Tensor = g(dx * dx)
         dy2: torch.Tensor = g(dy * dy)
         dxy: torch.Tensor = g(dx * dy)
-
         det_m: torch.Tensor = dx2 * dy2 - dxy * dxy
         trace_m: torch.Tensor = dx2 + dy2
 
         # compute the response map
-        scores: torch.Tensor = det_m - self.k * trace_m ** 2
+        scores: torch.Tensor = det_m - self.k * (trace_m ** 2)
 
         # threshold
         # TODO: add as signature parameter ?
-        scores = torch.clamp(scores, min=1e-6)
+        scores = torch.clamp(scores, min=0)
 
         # apply non maxima suppresion
-        scores = non_maxima_suppression2d(scores, kernel_size=(3, 3))
+        if self.nms:
+            scores = non_maxima_suppression2d(scores, kernel_size=(3, 3))
```

Comment:
should we pass kernel_size as parameter?

Reply:
For classical features, there is no such thing as non 3x3  nms, so I think, we should go for the constant. 
For the differentiable softnms - yes, but it will be in different PR